Add unique id for Scrape config entry entities

gjohansson-ST opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Use uuid to generate automatically unique id's for Scrape sensors made by config entry.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

gjohansson-ST committed:
scrape unique id

epenet commented on homeassistant/components/scrape/config_flow.py:
```diff
@@ -125,7 +127,15 @@ def validate_rest_setup(user_input: dict[str, Any]) -> dict[str, Any]:
+            {
+                **user_input,
+                CONF_INDEX: int(user_input[CONF_INDEX]),
+                CONF_UNIQUE_ID: uuid.uuid1(),
```
If we add UNIQUE_ID, then I think we need to remove DeviceClass, StateClass and UnitOfMeasurement.
We should probably also make it a string:
```suggestion
                CONF_UNIQUE_ID: str(uuid.uuid1()),
```

gjohansson-ST replied:
Why would we remove those?
Perhaps we should consider not allowing to change them (mess up statistics) but configure them initially we should do.

epenet replied:
If there is a unique_id, then they can be set in the frontend.
We can therefore default to None, and let the user customise afterwards (smaller config flow)

gjohansson-ST replied:
Looks like we should remove UoM but I don't see how a user would set DeviceClass nor StateClass?

epenet replied:
I must be mistaken - I can't find it either anymore.
We still need to ensure we store the unique_id as a string though.

epenet commented on tests/components/scrape/conftest.py:
```diff
@@ -78,3 +80,13 @@ async def load_integration(
+    """Automatically path uuid generator."""
+    with patch(
+        "homeassistant.components.scrape.config_flow.uuid.uuid1",
+        return_value="3699ef88-69e6-11ed-a1eb-0242ac120002",
```
```suggestion
        return_value=UUID("3699ef88-69e6-11ed-a1eb-0242ac120002"),
```

gjohansson-ST committed:
fix uuid str

epenet reviewed (commented):
I've changed my mind
I think it's better to keep unit of measurement for now.

gjohansson-ST commented:
> I've changed my mind I think it's better to keep unit of measurement for now.

Was also thinking a bit when I removed it. I also think it needs to stay.

gjohansson-ST committed:
add back UoM

epenet reviewed (approved):
Looks good!
Thanks @gjohansson-ST 👍
